contracts/VeJoeStaking.sol

```diff
@@ -305,7 +305,7 @@ contract VeJoeStaking is Initializable, OwnableUpgradeable {
 
         // Get the user's current veJOE balance and maximum veJOE they can hold
         uint256 userVeJoeBalance = veJoe.balanceOf(_user);
-        uint256 userMaxVeJoeCap = user.balance.mul(maxCap);
+        uint256 userMaxVeJoeCap = user.balance.mul(maxCapPct).div(100);
```

Comment:
note: I realized doing the calculation like this means that if `maxCapPct < 100`, e.g. `50`, the user would need to have at least `2` JOE staked but I think this is fine given that:

1. JOE and veJOE are 18 decimals. 
2. In practice, I don't think we will ever run into this case since I assume we are going to set `maxCapPct` to an initial value `>= 100` which means we won't be able to ever set it to be smaller than that
3. Even if we did want to handle this edge case, i think it would be difficult to handle it for logic in lines 313-314

curious if everyone agrees

Reply:
They wouldn't need 2 `JOE`, they would need 2e-18 `JOE` which is definitely fine.
This could be fixed with some basic math tho, but not needed.

Just for the record, something like that:
```
        uint256 userMaxVeJoeCap = user.balance.mul(maxCapPct); // this is 100 time bigger than expected
        if (userVeJoeBalance.mul(100) >= userMaxVeJoeCap) {
            // User already holds maximum amount of veJOE so there is no pending veJOE
            return 0;
        } else if (userVeJoeBalance.add(pendingVeJoe).mul(100) > userMaxVeJoeCap) {
            return userMaxVeJoeCap.sub(userVeJoeBalance.mul(100)).div(100);
        } else {
            return pendingVeJoe;
        }
```